Sources/GRPC/AsyncAwaitSupport/GRPCAsyncResponseStreamWriter.swift

```diff
@@ -19,26 +19,90 @@
 /// Writer for server-streaming RPC handlers to provide responses.
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
 public struct GRPCAsyncResponseStreamWriter<Response: Sendable>: Sendable {
+  public struct RecordingAsyncSequence: AsyncSequence {
+    public typealias Element = (Response, Compression)
+
+    private let stream: AsyncStream<(Response, Compression)>
+
+    init(stream: AsyncStream<(Response, Compression)>) {
+      self.stream = stream
+    }
+
+    public func makeAsyncIterator() -> AsyncIterator {
+      AsyncIterator(iterator: self.stream.makeAsyncIterator())
+    }
+
+    public struct AsyncIterator: AsyncIteratorProtocol {
+      private var iterator: AsyncStream<(Response, Compression)>.AsyncIterator
+
+      init(iterator: AsyncStream<(Response, Compression)>.AsyncIterator) {
+        self.iterator = iterator
+      }
+
+      public mutating func next() async -> Element? {
+        await self.iterator.next()
+      }
+    }
+  }
```

Comment:
Can we get a bunch of `@inlinable`s here?

Reply:
I don't think this stuff needs @inlinable it should only be used in tests where performance is moot.